### PR TITLE
[JENKINS-42846] - Rename slave.jar to agent.jar, keep symbolic link for compatibility

### PR DESCRIPTION
**Pull request by stopalopa**
Renamed the slave.jars to agent .jars.  I wasn't sure how to update the location of this image on docker hub to jenkins/agent.

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@stopalopa are you still interested in it?

**Comment by stopalopa** [comment]
Yeah, I'll try to add a symbolic link like you had suggested tonight.

**Comment by batmat** [comment]
@stopalopa the Dockerfile is conflicted FYI. Thanks!

File: Dockerfile [hunk 1]
```diff
@@ -32,13 +32,14 @@ ARG gid=1000
...
+LABEL Description="This is a base image, which provides the Jenkins agent executable (agent.jar)" Vendor="Jenkins project" Version="${VERSION}"
 
 ARG AGENT_WORKDIR=/home/${user}/agent
 
-RUN curl --create-dirs -fsSLo /usr/share/jenkins/slave.jar https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/${VERSION}/remoting-${VERSION}.jar \
+RUN curl --create-dirs -fsSLo /usr/share/jenkins/agent.jar https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/${VERSION}/remoting-${VERSION}.jar \
   && chmod 755 /usr/share/jenkins \
-  && chmod 644 /usr/share/jenkins/slave.jar
+  && chmod 644 /usr/share/jenkins/agent.jar
+RUN ln -sf /usr/share/jenkins/slave.jar /usr/share/jenkins/agent.jar
```
Comment (slide) [inline Dockerfile]: Can this be made part of the other RUN line?

**Comment by slide** [comment]
Conflicts are resolved. @stopalopa does this look ok to you?

**Comment by stopalopa** [comment]
@slide LGTM

**Comment by slide** [comment]
@batmat and @oleg-nenashev good to merge?

**Comment by oleg-nenashev** [comment]
Sorry, it fell between the cracks

**Comment by slide** [comment]
As soon as the build goes through, I think we can merge this. Thanks @stopalopa!

**Comment by oleg-nenashev** [comment]
🎆  , thanks all!

**Pull request merged**